codesign/lib/src/utils.dart

```diff
@@ -2,7 +2,42 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:io';
+
 import 'package:args/args.dart';
+import 'package:process/process.dart';
+
+/// helper function to generate unique next IDs.
+int id = 0;
+int get nextId {
+  final int currentKey = id;
+  id += 1;
+  return currentKey;
+}
+
+enum FILETYPE { FOLDER, ZIP, BINARY, OTHER }
+
+/// Check mime-type of file at [filePath] to determine if it is a directory.
+FILETYPE checkFileType(String filePath, ProcessManager processManager) {
+  final ProcessResult result = processManager.runSync(
+    <String>[
+      'file',
+      '--mime-type',
+      '-b', // is binary
+      filePath,
+    ],
+  );
+  String output = result.stdout as String;
+  if (output.contains('inode/directory')) {
```

Comment:
umm an enhanced enum which allows a constructor would require the constructor to be const, and therefore cannot have a constructor body that hold the if-else logic we have here. (https://stackoverflow.com/questions/38908285/how-do-i-add-methods-or-values-to-enums-in-dart). Shall we create a class, so that we can move the if-else logic into the constructor of the class?

Reply:
Unfortunately for you, it was in the last stable update to Dart :P

```dart
enum Test {
  a,
  b,
  c;
    
  factory Test.fromMimeType(String mimeType) {
    switch (mimeType) {
      case 'text/plain':
        return Test.a;
    }
    
    throw Exception('$mimeType does not exist for Test');
  }
}
```

Reply:
Thanks for pointing this out!